suitcase/fields.py

```diff
@@ -656,6 +725,51 @@ def setval(self, value):
         return self._get_parent_field().setval(value)
 
 
+class FieldArray(BaseField):
+    """Field which contains a list of some other field.
+
+    In some protocols, there exist repeated substructures which are present in a
+    variable number. The variable nature of these fields make a DispatchField
+    and DispatchTarget combination unsuitable; instead a FieldArray may be
+    used to pack/unpack these fields to/from a list.
+

```

Comment:
In fitting with the FieldArray format, I like this pattern

```
author = SubstructureField(PascalString16)
```

which omits the object instantiation. I'll work on writing up support for that today.